lib/screens/settings/sync_list_section.dart

```diff
@@ -46,7 +47,17 @@ class SyncListSection extends StatelessWidget {
                 value: peer.isServerRunning,
                 onChanged: kIsWeb
                     ? (value) => null
-                    : (value) => value ? peer.startServer() : peer.stopServer(),
+                    : (value) => value
+                        ? peer
+                            .startServer()
+                            .onError((error, stackTrace) => showDialog(
+                                  context: context,
+                                  builder: (context) => SimpleErrorPopupDialog(
+                                    'Could not start server',
+                                    error.toString(),
+                                  ),
+                                ))
+                        : peer.stopServer(),
```

Comment:
The `onChanged` expression is quite large now. Maybe it make sense to extract this into a own method inside the class. What do you think?